fsspec/spec.py

```diff
@@ -1425,17 +1465,51 @@ def from_json(blob):
         -------
         file system instance, not necessarily of this particular class.
         """
-        import json
+        return json.loads(blob, cls=AbstractFileSystem._JSONDecoder)
 
-        from .registry import _import_class, get_filesystem_class
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        JSON-serializable dictionary representation of this filesystem instance
 
-        dic = json.loads(blob)
-        protocol = dic.pop("protocol")
+        Returns
+        -------
+        Dictionary with keys cls (the python location of this class),
+        protocol (text name of this class's protocol, first one in case of
+        multiple), args (positional args, usually empty), and all other
+        kwargs as their own keys.
+        """
+        cls = type(self)
+        proto = self.protocol
+
+        return dict(
+            cls=f"{cls.__module__}.{cls.__name__}",
```

Comment:
Perhaps prefer with ":", which is slightly more readable?

Reply:
I copied this from the original code, but it also supports `module:class` format so I'll change it.